### PR TITLE
Client side configuration cleanup

### PR DESCRIPTION
**Pull request by ramsai1729**
* Removes unnecessary classes on client side
* Iterable<InetSocketAddress> is given in `ConnectionFactory.connect` passed down to `ClientConnectionEstablisher` - this helps in propagating topology changes

### CODE REVIEW

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java [hunk 1]
```diff
@@ -56,7 +53,7 @@
   private static final long                 MIN_RETRY_INTERVAL    = 1000;
   public static final String                RECONNECT_THREAD_NAME = "ConnectionEstablisher";
 
-  private final LinkedHashSet<ConnectionInfo>   connAddressProvider;
+  private volatile Iterable<InetSocketAddress> serverAddresses;
```
Comment (myronkscott) [inline common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: why is this volatile?  it is set in a synchronized block.

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java [hunk 2]
```diff
@@ -118,63 +114,56 @@ boolean getAllowReconnects() {
...
+        target = nextServerAddress(serverAddressIterator);
       }
       try {
-        return cmt.open(info);
+        return cmt.open(target);
       } catch (TransportRedirect redirect) {
-        ConnectionInfo add = new ConnectionInfo(redirect.getHostname(), redirect.getPort());
-        reporter.onError(info, redirect);
-        info = add;
-        if (this.connAddressProvider.add(add)) {
```
Comment (myronkscott) [inline common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: removing this causes redirects to be transient.  We want redirects to be permanent additions to the list of known servers.

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java (hunk 1 above)
Comment (ramsai1729) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: but not all reads under the synchronized block.

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java (hunk 2 above)
Comment (ramsai1729) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: this is not a problem If the user has specified all the servers in the stripe, right?

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java (hunk 1 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: toString reads don't matter.  I don't volatile is warranted here.

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java (hunk 2 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: true but that isn't going to happen all the time.

Comment (myronkscott) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: plus that ability was there before.  we should preserve it.

Comment (ramsai1729) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java:152]: Made some changes to preserve redirects, please have a look

File: common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java (hunk 1 above)
Comment (ramsai1729) [reply common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java]: what about reconnect(...)?

**Comment by ramsai1729** [comment]
@myronkscott PR rebased

**Pull request merged**